recirq/engine_utils.py

```diff
@@ -372,3 +375,39 @@ async def worker():
     await queue.join()
     for wjob in worker_jobs:
         wjob.cancel()
+
+
+def _get_current_time():
+    return datetime.datetime.now()
+
+
+def get_available_processors(processor_names: List[str]):
+    """Checks the reservation status of the processors and returns a list of
+    processors that are available to run on at the present time.
+    """
+    project_id = os.environ['GOOGLE_CLOUD_PROJECT']
+    engine = cirq.google.get_engine()
+    available_processors = []
+    for processor_name in processor_names:
+        processor_id = get_processor_id_by_device_name(processor_name)
+        if processor_id is None:
+            # Skip the check if this is a simulator.
+            continue
+        processor = engine.get_processor(processor_id)
+        for time_slot in processor.get_schedule():
+            try:
+                time_slot = EngineTimeSlot.from_proto(time_slot)
+            # Parsing the end_time of the last time range in the schedule might give throw an error.
```

Comment:
Is this expected?   This seems like a problem.  I am not sure we should just ignore ValueError blindly.

Reply:
It seems to be a problem in the EngineTimeSlot.from_proto function in cirq. For example, the end time of the last interval in the schedule has 
```
end_time {                                                                                                                                  
  seconds: 253402300799
}
```
which raises `ValueError: year 0 is out of range` when trying to parse it with the datetime library.

Reply:
Ok, I will go and fix this in cirq.  Let's add a TODO (and ideally, a github issue) to go back and fix this.

Reply:
Added a TODO, and created an issue in cirq: https://github.com/quantumlib/Cirq/issues/3787